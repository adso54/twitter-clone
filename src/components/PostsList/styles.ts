import styled from 'styled-components'
import {generateStylesOnAllResolution} from '../../utils/mediaQueries'

const StyledPostsList = generateStylesOnAllResolution( 
    `flex-grow: 2;
    grid-area: postslist;`,
    {
        tablet:``,
        laptop:``,
        widescreen:``
    },
    styled.div
)

export {StyledPostsList}