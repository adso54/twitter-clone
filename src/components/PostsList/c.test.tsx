import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import PostsList from './index'

test('Rendered PostsList match the snapshot', async () => {
    const testPostListData: PostsList = {
        postsList: [
            {
              userImage: "https://picsum.photos/100/100",
              username: "Jan",
              twitterName: "@JanK",
              date: "03/19/2021",
              content: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.",
              comments: 25,
              retweets: 20,
              likes: 430,
              postId: "ff8031cc-b4ba-47fd-aaf7-8ef80b379a14"
            },
            {
              userImage: "https://picsum.photos/100/100",
              username: "Anna",
              twitterName: "@AnnaK",
              date: "01/22/2021",
              content: "It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).",
              comments: 53,
              retweets: 30,
              likes: 230,
              postId: "40f02540-2570-493e-8aee-baefd3587d82"
            }]
    }
    const component = render(PostsList(testPostListData))
    expect(component).toMatchSnapshot()
})