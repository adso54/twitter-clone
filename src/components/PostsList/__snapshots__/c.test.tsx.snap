// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendered PostsList match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ hBkaYh"
      >
        <div
          class="sc-fEOsli cGPlUY"
          id="ff8031cc-b4ba-47fd-aaf7-8ef80b379a14"
        >
          <image
            class="sc-bjUoiL iriXeT"
          >
            <img
              alt="@JanK"
              src="https://picsum.photos/100/100"
            />
          </image>
          <div
            class="sc-idiyUo huHBJj"
          >
            <div
              class="sc-dIouRR gsBpoC"
            >
              <div
                class="sc-hHLeRK bPncNK"
              >
                Jan
              </div>
              <div
                class="sc-dmRaPn hzTlRs"
              >
                @JanK
              </div>
              <div
                class="sc-kgflAQ fHLEnO"
              >
                03/19/2021
              </div>
            </div>
            <div
              class="sc-fLlhyt gIUZPJ"
            >
              Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
            </div>
            <div
              class="sc-bBrHrO eUFZoM"
            >
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  comment.svg
                </svg>
                <div>
                  25
                </div>
              </div>
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  share.svg
                </svg>
                <div>
                  20
                </div>
              </div>
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  heart.svg
                </svg>
                <div>
                  430
                </div>
              </div>
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  download.svg
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="sc-fEOsli cGPlUY"
          id="40f02540-2570-493e-8aee-baefd3587d82"
        >
          <image
            class="sc-bjUoiL iriXeT"
          >
            <img
              alt="@AnnaK"
              src="https://picsum.photos/100/100"
            />
          </image>
          <div
            class="sc-idiyUo huHBJj"
          >
            <div
              class="sc-dIouRR gsBpoC"
            >
              <div
                class="sc-hHLeRK bPncNK"
              >
                Anna
              </div>
              <div
                class="sc-dmRaPn hzTlRs"
              >
                @AnnaK
              </div>
              <div
                class="sc-kgflAQ fHLEnO"
              >
                01/22/2021
              </div>
            </div>
            <div
              class="sc-fLlhyt gIUZPJ"
            >
              It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).
            </div>
            <div
              class="sc-bBrHrO eUFZoM"
            >
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  comment.svg
                </svg>
                <div>
                  53
                </div>
              </div>
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  share.svg
                </svg>
                <div>
                  30
                </div>
              </div>
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  heart.svg
                </svg>
                <div>
                  230
                </div>
              </div>
              <div
                class="sc-ivTmOn dZwmUq"
              >
                <svg>
                  download.svg
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ hBkaYh"
    >
      <div
        class="sc-fEOsli cGPlUY"
        id="ff8031cc-b4ba-47fd-aaf7-8ef80b379a14"
      >
        <image
          class="sc-bjUoiL iriXeT"
        >
          <img
            alt="@JanK"
            src="https://picsum.photos/100/100"
          />
        </image>
        <div
          class="sc-idiyUo huHBJj"
        >
          <div
            class="sc-dIouRR gsBpoC"
          >
            <div
              class="sc-hHLeRK bPncNK"
            >
              Jan
            </div>
            <div
              class="sc-dmRaPn hzTlRs"
            >
              @JanK
            </div>
            <div
              class="sc-kgflAQ fHLEnO"
            >
              03/19/2021
            </div>
          </div>
          <div
            class="sc-fLlhyt gIUZPJ"
          >
            Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.
          </div>
          <div
            class="sc-bBrHrO eUFZoM"
          >
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                comment.svg
              </svg>
              <div>
                25
              </div>
            </div>
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                share.svg
              </svg>
              <div>
                20
              </div>
            </div>
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                heart.svg
              </svg>
              <div>
                430
              </div>
            </div>
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                download.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="sc-fEOsli cGPlUY"
        id="40f02540-2570-493e-8aee-baefd3587d82"
      >
        <image
          class="sc-bjUoiL iriXeT"
        >
          <img
            alt="@AnnaK"
            src="https://picsum.photos/100/100"
          />
        </image>
        <div
          class="sc-idiyUo huHBJj"
        >
          <div
            class="sc-dIouRR gsBpoC"
          >
            <div
              class="sc-hHLeRK bPncNK"
            >
              Anna
            </div>
            <div
              class="sc-dmRaPn hzTlRs"
            >
              @AnnaK
            </div>
            <div
              class="sc-kgflAQ fHLEnO"
            >
              01/22/2021
            </div>
          </div>
          <div
            class="sc-fLlhyt gIUZPJ"
          >
            It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using 'Content here, content here', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for 'lorem ipsum' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).
          </div>
          <div
            class="sc-bBrHrO eUFZoM"
          >
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                comment.svg
              </svg>
              <div>
                53
              </div>
            </div>
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                share.svg
              </svg>
              <div>
                30
              </div>
            </div>
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                heart.svg
              </svg>
              <div>
                230
              </div>
            </div>
            <div
              class="sc-ivTmOn dZwmUq"
            >
              <svg>
                download.svg
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
